Asterisks in recipe lists are not wrapped as strings

<!--
Thank you for reporting an issue with OpenRewrite!
We appreciate you taking the time to help us improve.
Please fill out the template below to help us understand and reproduce the issue.
Feel free to delete any sections that don't apply to your issue.
-->

## What version of OpenRewrite are you using?
<!--
Whenever possible please try to replicate your issue with the latest versions of OpenRewrite.
The latest major and minor versions of OpenRewrite projects are usually listed here:
https://docs.openrewrite.org/reference/latest-versions-of-every-openrewrite-module
For patch releases check the GitHub Releases page for the respective project.

We release every few weeks, so it's possible that your issue has already been fixed.

If you want to try the most recent changes that haven't been fully released yet, you can check out our snapshot releases
https://docs.openrewrite.org/reference/snapshot-instructions
-->
I am using

- Maven/Gradle plugin v6.1.3

<!--
Are you using the Maven plugin, Gradle plugin, Moderne CLI, Moderne.io or something else?
Is your project a single module or a multi-module project?

Can you share your configuration so that we can rule out any configuration issues?

Is your project public? If so, can you share a link to it?
Code snippets can also be shared privately via [our public Slack](https://join.slack.com/t/rewriteoss/shared_invite/zt-nj42n3ea-b~62rIHzb3Vo0E1APKCXEA).
-->
The recipe list for something like the [Spring Boot 3.3 upgrade](https://docs.openrewrite.org/recipes/java/spring/boot3/upgradespringboot_3_3) has sections like this:

```yaml
recipeList:
  - org.openrewrite.java.spring.boot3.UpgradeSpringBoot_3_2
  - org.openrewrite.java.spring.boot3.SpringBootProperties_3_3
  - org.openrewrite.java.dependencies.UpgradeDependencyVersion:
      groupId: org.springframework.boot
      artifactId: *
      newVersion: 3.3.x
      overrideManagedVersion: false
```

The `artifactId` is a `*`. When copying this YAML into a custom recipe, the Maven plugin fails:

```shell
[ERROR] Failed to execute goal org.openrewrite.maven:rewrite-maven-plugin:6.1.3:run (default-cli) on project asterisk: Execution default-cli of goal org.openrewrite.maven:rewrite-maven-plugin:6.1.3:run failed: while scanning an alias
[ERROR]  in 'reader', line 14, column 19:
[ERROR]           artifactId: *
[ERROR]                       ^
[ERROR] unexpected character found
[ERROR] (10)
[ERROR]  in 'reader', line 14, column 20:
[ERROR]           artifactId: *
[ERROR]                        ^
[ERROR]
[ERROR] -> [Help 1]
```

The [source of the recipe](https://github.com/openrewrite/rewrite-spring/blob/main/src/main/resources/META-INF/rewrite/spring-boot-33.yml) has the asterisk wrapped as a string. 


## Are you interested in [contributing a fix to OpenRewrite](https://github.com/openrewrite/.github/blob/main/CONTRIBUTING.md#contributing-fixes)?
<!-- Indicate if this is something you would like to work on, and how we can best support you in doing so. -->

Yes, I can open a PR. The proposed fix would be to handle the asterisk as a special case inside `RecipeDescriptorExtension.kt`:

```kotlin
val formattedValue = if (value is Array<*>) {
        val asArray =  value as Array<*>
        "[${asArray.joinToString(", ")}]"
    } else if (value == "*") {
        "\"*\""
    } else {
        value
    }
```

This produces the expected output:

```yaml
recipeList:
  - org.openrewrite.java.spring.boot3.UpgradeSpringBoot_3_2
  - org.openrewrite.java.spring.boot3.SpringBootProperties_3_3
  - org.openrewrite.java.dependencies.UpgradeDependencyVersion:
      groupId: org.springframework.boot
      artifactId: "*"
      newVersion: 3.3.x
      overrideManagedVersion: false
```

--- a/src/main/kotlin/org/openrewrite/RecipeDescriptorExtensions.kt
+++ b/src/main/kotlin/org/openrewrite/RecipeDescriptorExtensions.kt
@@ -59,6 +59,8 @@ fun OptionDescriptor.asYaml(indentation: Int = 0): String {
     val formattedValue = if (value is Array<*>) {
         val asArray =  value as Array<*>
         "[${asArray.joinToString(", ")}]"
+    } else if (value == "*") {
+        "\"*\""
     } else {
         value
     }

